mono/metadata/object.c

```diff
@@ -3966,25 +3967,25 @@ mono_runtime_get_main_args (void)
  * \returns a \c MonoArray with the arguments passed to the main
  * program. On failure returns NULL and sets \p error.
  */
-MonoArray*
-mono_runtime_get_main_args_checked (MonoError *error)
+MonoArrayHandle*
+mono_runtime_get_main_args_checked_handle (MonoError *error)
```

Comment:
We usually return handles by value.  ie: `MonoArrayHandle mono_runtime_get_main_args_handle (...)` 

(I usually try not to stack `_checked` & `_handle` suffixes - my assumption is that anything with a `_handle` should be using `MonoError*` for error reporting)